Source/Core/Core/Boot/Boot_BS2Emu.cpp

```diff
@@ -209,7 +211,20 @@ bool CBoot::EmulatedBS2_GC(const DiscIO::VolumeDisc& volume)
 
   SetupGCMemory();
 
-  DVDRead(volume, /*offset*/ 0x00000000, /*address*/ 0x00000000, 0x20, DiscIO::PARTITION_NONE);
+  DVDReadDiscID(volume, 0x00000000);
+
+  bool streaming = Memory::Read_U8(0x80000008);
+  if (streaming)
+  {
+    u8 streaming_size = Memory::Read_U8(0x80000009);
+    if (streaming_size == 0)
+      streaming_size = 10;
```

Comment:
What's so special about zero (and ten, for that matter)?

Reply:
BS2 itself makes that check.  If byte 9 (streaming size) on the disc is set to 0, it treats the size as 10; otherwise, it uses that size.  I documented it on the [Wii Disc](https://wiibrew.org/wiki/Wii_Disc) and it was also already documented on the [BS2 page](https://wiibrew.org/wiki/System_Menu/BS2) for state 11, albeit not very explicitly:

> if not waiting for a disk, exit to state 50
> 
> if waiting, disable the waiting-for-disk and check streaming [0x80000008]
> 
> if streaming and no streaming buffer size defined [0x80000009], set to size 10
> 
> if streaming, setup via __DVDAudioBufferConfig
> 
> go to state 12 

To my understanding, no disc actually specifies a buffer size other than 0, so it probably doesn't come up in practice.  I'll add a comment explaining it.

It's worth noting that this check happens in both Wii and GC BS2, and on the Wii it only happens for Gamecube discs (to my understanding, Dolphin doesn't use the emulated wii bs2 when running GC discs).

There's some additional that prints a message out in the Wii Menu (separate from the code that configured the audio buffer which happens earlier).  In 4.3U it's located from 81379ab8 through 81379b54.  Tidied, it looks like this:

```C
OSReport("\n"); // 8169653f
OSReport("  Game Name ... %-4.4s\n",DiscHeader.GameID);      // 81645842, 80000000
OSReport("  Company ..... %-2.2s\n",DiscHeader.MakerID);     // 8164585a, 80000004
OSReport("  Disk # ...... %x\n",    DiscHeader.DiscNumber);  // 81645872, 80000006
OSReport("  Game ver .... %x\n",    DiscHeader.DiscVersion); // 81645886, 80000007
char * streamingEnabledStr = "OFF"; // 81696554
if (DiscHeader.DiscStreamingFlag != 0) { // 80000008
  streamingEnabledStr = "ON"; // 81696551
}
OSReport("  Streaming ... %s\n",streamingEnabledStr); // 816458db
if (DiscHeader.DiscStreamingFlag != 0) { // 80000008
  if (DiscHeader.DiscStreamingBufferSize == 0) { // 80000009
    OSReport("    Buf size .. 10(default)\n"); // 81645903
  }
  else {
    OSReport("    Buf size .. %d\n",DiscHeader.DiscStreamingBufferSize); // 816458ef, 80000009
  }
}
```